src/controllers/mainController.ts

```diff
@@ -747,9 +749,13 @@ export default class MainController implements vscode.Disposable {
             if (editor.document.getText(selectionToTrim).trim().length === 0) {
                 return;
             }
-            await self._outputContentProvider.runQuery(self._statusview, uri, querySelection, title);
+            vscode.commands.executeCommand('setContext', 'mssql.isQueryRunning', true);
+            await self._outputContentProvider.runQuery(self._statusview, uri, querySelection, title).then(() => {
+                vscode.commands.executeCommand('setContext', 'mssql.isQueryRunning', false);
```

Comment:
You're already awaiting this, you don't need to put it in a .then

But regardless, it would be easier just to remove this and the one on 758 and put it inside a finally block instead.

Reply:
Apparently, the code seems to ignore the await and the finally block executes before the query is complete (in the case of an infinite query that is). This means it is not working as expected.